middle_end/flambda2/from_lambda/lambda_to_flambda.ml

```diff
@@ -351,7 +351,7 @@ let transform_primitive env (prim : L.primitive) args loc =
                cond,
                arg1,
                switch_for_if_then_else ~cond:(L.Lvar cond)
-                 ~ifso:(L.Lvar const_true) ~ifnot:arg2 ) ))
+                 ~ifso:(L.Lvar const_true) ~ifnot:arg2 ~kind:Pgenval ) ))
```

Comment:
Here (and below for `Psequand`) the result is a boolean, so `Pintval` should be correct.